File: 10828.py
# https://www.acmicpc.net/problem/10828
# 스택

from collections import deque
import sys

dq = deque()

n = int(input())
for _ in range(n):
    query = list(sys.stdin.readline().split())
    if query[0] == "push":
        num = int(query[1])
        dq.append(num)
    elif query[0] == "pop":
        if len(dq):
            print(dq.pop())
        else:
            print(-1)
    elif query[0] == "top":
        if len(dq):
            print(dq[len(dq) - 1])
        else:
            print(-1)
    elif query[0] == "size":
        print(len(dq))
    elif query[0] == "empty":
        if len(dq):
            print(0)
        else:
            print(1)

File: 10828.cpp
// https://www.acmicpc.net/problem/10828
// 스택

#include <iostream>
#include <stack>
using namespace std;

int main() {
	int n;
	string query;
	int input_num;
	stack<int> s;

	cin >> n;

	for (int i = 0; i < n; ++i) {
		cin >> query;

		if (query == "push") {
			cin >> input_num;
			s.push(input_num);
		} else if (query == "pop") {
			if (!s.empty()) {
				cout << s.top() << endl;
				s.pop();
			} else {
				cout << -1 << endl;
			}
		} else if (query == "top") {
			if (!s.empty()) {
				cout << s.top() << endl;
			} else {
				cout << -1 << endl;
			}
		} else if (query == "size") {
			cout << s.size() << endl;
		} else if (query == "empty") {
			cout << s.empty() << endl;;
		}
	}

	return 0;
}
